awscli/customizations/codeartifact/login.py

```diff
@@ -84,13 +84,16 @@ def _run_commands(self, tool, commands, dry_run=False):
 
     def _run_command(self, tool, command, *, ignore_errors=False):
         try:
-            self.subprocess_utils.check_call(
+            self.subprocess_utils.run(
                 command,
-                stdout=self.subprocess_utils.PIPE,
-                stderr=self.subprocess_utils.PIPE
+                capture_output=True,
+                check=True
             )
         except subprocess.CalledProcessError as ex:
             if not ignore_errors:
+                sys.stdout.write(
+                    ex.stderr.decode(os.device_encoding(1) or "utf-8")
```

Comment:
For determining encoding of stderr, I think it would be worth exposing a `get_stderr_encoding` in the `compat.py` module similar to what we have here: https://github.com/aws/aws-cli/blob/9abc05dc392d5466b0d728006b177db10f4fe4cb/awscli/testutils.py#L654. The advantage in going with this approach is that in accessing `sys.__stderr__` we definitively know what standard error is going to be from the start of the Python program.